File: backend/src/main/java/com/smartlife/model/Task.java
package com.smartlife.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskStatus status = TaskStatus.TODO;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    private Priority priority = Priority.MEDIUM;

    @Builder.Default
    @Column(length = 30)
    private String category = "PERSONAL";

    private LocalDateTime startDate;

    private LocalDateTime dueDate;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum TaskStatus { TODO, IN_PROGRESS, DONE }
    public enum Priority { LOW, MEDIUM, HIGH }
}

File: mobile-android/app/src/main/java/com/smartlife/mobile/feature/tasks/data/model/Task.kt
package com.smartlife.mobile.feature.tasks.data.model

data class Task(
    val id: Long = 0,
    val title: String = "",
    val description: String? = null,
    val category: String = "PERSONAL",
    val priority: String = "MEDIUM",
    // Backend renvoie `status` (TODO/IN_PROGRESS/DONE), pas un booléen `completed`.
    val status: String = "TODO",
    val startDate: String? = null,
    val dueDate: String? = null,
    val createdAt: String = "",
) {
    val completed: Boolean get() = status == "DONE"
}

data class CreateTaskRequest(
    val title: String,
    val category: String = "PERSONAL",
    val priority: String = "MEDIUM",
    val status: String = "TODO",
    val description: String? = null,
    val dueDate: String? = null,
)
